src/biom.cpp

```diff
@@ -77,30 +112,55 @@ biom::biom(std::string filename) {
                 sizeof(unsigned int) * n_obs, __FILE__, __LINE__);
         exit(EXIT_FAILURE);
     }
+}
+
+biom::biom() { 
```

Comment:
I would suggest  explicitly initialize has_hdf5_backing=false, since several functions look for it.